tools/generate-test-cases

```diff
@@ -3,84 +3,75 @@
+    "ext4-filesystem": "nspawn",
+    "tar": "nspawn-extract",
+    "openstack": "qemu",
+    "qcow2": "qemu",
+    "vhd": "qemu",        
+    "vmdk": "qemu",
+    "ami": "qemu-extract",
 }
 
-It then outputs a json test case as standard output. 
-'''
-
-
-def run_osbuild(manifest, store):
-    osbuild_cmd = ["osbuild", "--store", store, "--json", "-"]
-    result = json.loads(subprocess.check_output(osbuild_cmd, encoding="utf-8", input=json.dumps(manifest)))
-    return result.get("output_id")
-
-
-def get_boot_method(output_format):
-    format_boot_dict = {
-        "partitioned-disk": "nspawn",
-        "ext4-filesystem": "nspawn",
-        "tar": "nspawn-extract",
-        "openstack": "qemu",
-        "qcow2": "qemu",
-        "vhd": "qemu",        
-        "vmdk": "qemu",
-        "ami": "qemu-extract",
-    }
-    return format_boot_dict[output_format]
-
-
-def main(compose_request, store):
-    boot = get_boot_method(compose_request["output-format"])
-    test_case = {
-        "boot": {
-            "type": boot
-        },
-        "compose-request": compose_request
-    }
-
-    blueprint = json.dumps(compose_request["blueprint"])
-
-    pipeline_command = ["go", "run", "./cmd/osbuild-pipeline", "-distro", compose_request["distro"], "-arch", compose_request["arch"], "-image-type", compose_request["output-format"], "-"]
-    test_case["manifest"] = json.loads(subprocess.check_output(pipeline_command, input=blueprint, encoding="utf-8"))
-
-    pipeline_command = ["go", "run", "./cmd/osbuild-pipeline", "-distro", compose_request["distro"], "-arch", compose_request["arch"], "-image-type", compose_request["output-format"], "-rpmmd", "-"]
-    test_case["rpmmd"] = json.loads(subprocess.check_output(pipeline_command, input=blueprint, encoding="utf-8"))
-
-    if boot != "nspawn-extract":
-        output_id = run_osbuild(test_case["manifest"], store)
-        image_file = os.path.join(store, "refs", output_id, compose_request["filename"])
-
-        # we don't yet support image-info on directory trees
-        if boot == "qemu-extract":
-            fn, ex = os.path.splitext(image_file)
-            if ex == ".xz":
-                with open(fn, "w") as f:
-                    subprocess.run(["xz", "--decompress", "--stdout", image_file], stdout=f)
-                image_file = fn
-        test_case["image-info"] = json.loads(subprocess.check_output(["tools/image-info", image_file]))
+format_name_dict = {
+    "partitioned-disk": "disk.img",
+    "ext4-filesystem": "filesystem.img",
+    "tar": "root.tar.xz",
+    "openstack": "disk.qcow2",
+    "qcow2": "disk.qcow2",
+    "vhd": "disk.vhd",        
+    "vmdk": "disk.vmdk",
+    "ami": "image.raw.xz",
+}
 
-    return test_case
 
+def main(distro, arch, store, output):
+    test_case = []
+    for output_format, boot in format_boot_dict.items():
+        test_case_request = {
+            "boot": {
+                "type": boot
+            },
+            "compose-request": {
```

Comment:
I don't think this works. i think you need a map from image type to each test_case_request, as the blueprints will be different. AMI uses an empty blueprint, the others have the ssh key, and some have the openssh-server package explicitly enabled. Running the script should not change the actual test cases, but this does, right? (could add some new fields to rpmmd, but that's not crucial as they are unused).
Overall this patch looks great though!

Reply:
I now only use a single mapping between the test case request and the output type. The test case requests are located in the format-request-map.json.